internal/migrations/migrator.go

```diff
@@ -349,45 +330,62 @@ func (m *Migrator) PersistLastChangesetHeight(ctx context.Context, tx sql.Execut
 	return setLastStoredChangeset(ctx, tx, m.lastChangeset)
 }
 
-func (m *Migrator) MigrationStatus() types.MigrationStatus {
-	m.mu.RLock()
-	defer m.mu.RUnlock()
-
-	if m.activeMigration == nil {
-		return types.NoActiveMigration
-	}
+func (m *Migrator) MigrationStatus(ctx context.Context) (types.MigrationStatus, error) {
+	if m.genesisMigrationParams.EndHeight == 0 {
```

Comment:
I guess this seems ok for now. We make a lot of unnecessary DB calls here. A lot of this info is held in memory by ABCI, but we are forced to retrieve it from storage.

This is a larger problem with migrator, so we do not have to fix for now IMO

Reply:
Fixed this. DB calls occur only in the constructor. and rest of the time, its held internally by the migrator.

Reply:
wdym? I don't think this is fixable, since this data isn't held in memory.

Reply:
just pushed the changes

Reply:
sorry, more changes on the way